src/components/structures/ScrollPanel.js

```diff
@@ -551,6 +551,11 @@ module.exports = React.createClass({
         var messages = this.refs.itemlist.children;
         for (var i = messages.length-1; i >= 0; --i) {
             var m = messages[i];
+            if (m.dataset.containedScrollTokens &&
```

Comment:
let's have some comments about what's going on here.

Note also that the docs at the top describe `data-scroll-token`, and need an update.